### PR TITLE
Feature(messaging): settings APIs

### PR DESCRIPTION
**Pull request by SamSalvatico**
Description

Update APIs and SDKs for users settings

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [x] **New feature**
- [ ] **Dev change**

### CODE REVIEW

File: apps/messaging-api/services/users/invitations/accept-invitations.ts [hunk 1]
```diff
@@ -28,6 +31,35 @@ export const getInvitationForUser = async (params: {
...
+      userProfileId,
+      errorCode: ACCEPT_INVITATIONS_ERROR,
+    });
+    if (!user.userProfileId) {
+      throw createError(
+        ACCEPT_INVITATIONS_ERROR,
+        "The current user has no related user profile",
+        400,
+      )();
+    }
```
Comment (Thurfjell) [inline apps/messaging-api/services/users/invitations/accept-invitations.ts:52]: Why is this needed? 
getUserInvitations seems to require the profile id. Should it not be able to determine based on this value and the credentials, what you should get in response?

File: apps/messaging-api/services/users/invitations/accept-invitations.ts [hunk 2]
```diff
@@ -158,6 +190,42 @@ export const updateInvitationStatus = async (params: {
   }
 };
 
+export const getInvitationStatus = async (params: {
+  pg: PostgresDb;
+  userProfileId: string;
+}): Promise<{ userStatus: string }> => {
+  const client = await params.pg.connect();
```
Comment (Thurfjell) [inline apps/messaging-api/services/users/invitations/accept-invitations.ts:197]: `pg.pool.connect()` ?

File: apps/messaging-api/services/users/invitations/accept-invitations.ts [hunk 3]
```diff
@@ -158,6 +190,42 @@ export const updateInvitationStatus = async (params: {
...
+    throw toOutput;
+  } finally {
+    client.release();
+  }
+
+  if (statusResponse && statusResponse.rowCount) {
+    return statusResponse.rows[0];
+  }
+
+  throw createError(ACCEPT_INVITATIONS_ERROR, "Cannot find the user", 404)();
```
Comment (Thurfjell) [inline apps/messaging-api/services/users/invitations/accept-invitations.ts:226]: I think you should revert this. 

We want the happy path at the end of the function, not the catastrophic one no? 

eg.

```
...
if(fail) throw
...
return happy
```

File: apps/messaging-api/services/users/invitations/accept-invitations.ts (hunk 1 above)
Comment (SamSalvatico) [reply apps/messaging-api/services/users/invitations/accept-invitations.ts:52]: You're totally right 🤦

**Review by Thurfjell — APPROVED** [review]
